### PR TITLE
Vastly rewrite and expand `et/README.md`

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/157876.

/cc @dcharkes @jensjoha as well (documented `--gn-args`)

### CODE REVIEW

File: tools/engine_tool/README.md [hunk 1]
```diff
@@ -1,66 +1,337 @@
...
+Similar to formatters, global linters can be run with `et lint`:
+
+```sh
+et lint
+```
+
+At the time of this writing, the linters always operate on the entire
+repository.
+
+### Running a Flutter app with a local engine build
```
Comment (matanlurey) [inline tools/engine_tool/README.md:236]: TODO: Add this before I merge.

**Review by jtmcdole — APPROVED** [review]
LGTM.

Small ask: a caution about building multiple engines and consuming a lot of space. The easier we make it for folks to build, the quicker they'll lose their free space.

File: tools/engine_tool/README.md [hunk 2]
```diff
@@ -1,66 +1,337 @@
...
+  format   Formats files using standard formatters and styles.
+  lint     Lint the engine repository.
+  query    Provides information about build configurations and tests.
+  run      Run a Flutter app with a local engine build.
+  test     Runs a test target
+
+Run "et help <command>" for more information about a command.
+```
+
+## Understanding the concept of a build configuration
```
[the remark this answers is not in the document]
Comment (matanlurey) [reply tools/engine_tool/README.md:78]: Done.

[the remark this answers is not in the document]
Comment (matanlurey) [reply tools/engine_tool/README.md:78]: Done, added as a TIP below.

File: tools/engine_tool/README.md [hunk 3]
```diff
@@ -1,66 +1,337 @@
...
+
+## Common Tasks
+
+Tasks we expect the majority of contributors and users of the engine to need.
+
+### Building a host engine
+
+The default and most common operation is to build a _host_ variant of the
+engine, or an engine that runs on the local (desktop) operating system currently
+being used. For example, when running on an ARM64 MacOS laptop, an ARM64 MacOS
```
[the remark this answers is not in the document]
Comment (matanlurey) [reply tools/engine_tool/README.md]: Done.

File: tools/engine_tool/README.md (hunk 1 above)
Comment (matanlurey) [reply tools/engine_tool/README.md:243]: Done.

**Comment by matanlurey** [comment]
> LGTM.
> 
> Small ask: a caution about building multiple engines and consuming a lot of space. The easier we make it for folks to build, the quicker they'll lose their free space.

Done!

**Review by zanderso — APPROVED** [review]
Thanks for improving these docs!

**Pull request merged**